security/pkg/server/ca/authenticate/kube_jwt.go

```diff
@@ -69,9 +76,13 @@ func (a *KubeJWTAuthenticator) Authenticate(ctx context.Context) (*Caller, error
 	if err != nil {
 		return nil, fmt.Errorf("target JWT extraction error: %v", err)
 	}
-	id, err := a.client.ValidateK8sJwt(targetJWT, a.jwtPolicy)
-	if err != nil {
-		return nil, fmt.Errorf("failed to validate the JWT: %v", err)
+	id, err := a.client.ValidateK8sJwt(targetJWT, a.jwtPolicy, a.clusterID)
+	if err != nil || len(id) != 2 {
```

Comment:
Falling back to a remote cluster for the verification is fine for the first version of implementation.
I think in the future we can inspect the issuer of the JWT and directly verify it against the mapped cluster. This requires us to maintain a mapping between the clusters and the issuer IDs.

Reply:
Thanks, I will address this in a following up PR